.github/workflows/main.yml

```diff
@@ -8,20 +8,30 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout sources
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Install stable toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          override: true
-          components: rustfmt
+        run: |
+          rustup component add rustfmt
 
       - name: Check Formatting
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+        run: |
+          cargo fmt --all -- --check
+
+  clippy:
+    # needs: [test]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v4
+      - run: |
+          rustup set auto-self-update disable
+          rustup toolchain install nightly -c clippy
+          rustup default nightly
```

Comment:
Hmm why are you suggesting that we use `nightly` for clippy? 🤔

Reply:
I'm sorry. I was thinking nightly clippy sometimes has better lints. But maybe bringing nightly is unneeded for the task. I've switched to stable instead.

Reply:
Yeah you're right. It might have better lints, but it might also have incompatible lints to the current stable rust version that we're using too 😄